axlearn/open_api/metrics/tool_use_execution_utils.py

```diff
@@ -0,0 +1,307 @@
+# Copyright © 2024 Apple Inc.
+"""Utilities for the detailed tool use metrics."""
+
+import re
+import string
+from typing import Dict, List, Union
+
+from typing_extensions import TypeAlias
+
+Value = Union[str, int, bool, float]
+ValueOrListOf: TypeAlias = Union[Value, List[Value]]
```

Comment:
```suggestion
ValueOrListOf: TypeAlias = Union[Value, list[Value]]
```
nit -- we are trying to move away from the deprecated typing annotations.

Reply:
Done